src/TurnerSoftware.SitemapTools/Parser/TextSitemapParser.cs

```diff
@@ -1,20 +1,22 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.IO;
-using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace TurnerSoftware.SitemapTools.Parser
 {
 	public class TextSitemapParser : ISitemapParser
 	{
-		public async Task<SitemapFile> ParseSitemapAsync(TextReader reader)
+		public async Task<SitemapFile> ParseSitemapAsync(TextReader reader, CancellationToken cancellationToken = default)
 		{
 			var sitemapEntries = new List<SitemapEntry>();
 
 			string line;
 			while ((line = await reader.ReadLineAsync()) != null)
 			{
+				if (cancellationToken.IsCancellationRequested)
+					throw new OperationCanceledException();
```

Comment:
Rather than throwing it explicitly like this, we can call `cancellationToken.ThrowIfCancellationRequested();` instead - it will do the if-statement and throw for us. If you could change them all to that, that would be great.

Reply:
Thanks.